Last modified by Soshant Bali on 11/30/2016

In [1]:
import pandas as pd
import numpy as np
import sklearn as sk
import statsmodels.api as sm
import matplotlib.pyplot as plt

In [12]:
#Generate data
x=np.random.normal(0,1,100)
y=2*x + np.random.normal(0,1,100)
#convert numpy arrays to pandas data frame
xy_df = pd.DataFrame({'y':y,'x':x})

In [13]:
#Generalized linear model (no intercept)
model = sm.formula.ols(formula='y ~ x + 0',data=xy_df)
res = model.fit()
print (res.summary())

                            OLS Regression Results                            
Dep. Variable:                      y   R-squared:                       0.785
Model:                            OLS   Adj. R-squared:                  0.783
Method:                 Least Squares   F-statistic:                     361.8
Date:                Wed, 30 Nov 2016   Prob (F-statistic):           7.87e-35
Time:                        18:30:17   Log-Likelihood:                -141.77
No. Observations:                 100   AIC:                             285.5
Df Residuals:                      99   BIC:                             288.2
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [95.0% Conf. Int.]
------------------------------------------------------------------------------
x              1.9529      0.103     19.020      0.0

(a) Coefficient estimate (slope) is 1.9529, and standard error of this estimate is 0.103. Null hypothesis that slope is zero is rejected, because, t-statistic is 19.02, & p-value 0 (or very low). 

In [15]:
#Generalized linear model (with intercept)
model = sm.formula.ols(formula='x ~ y + 0',data=xy_df)
res = model.fit()
print (res.summary())

                            OLS Regression Results                            
Dep. Variable:                      x   R-squared:                       0.785
Model:                            OLS   Adj. R-squared:                  0.783
Method:                 Least Squares   F-statistic:                     361.8
Date:                Thu, 01 Dec 2016   Prob (F-statistic):           7.87e-35
Time:                        10:07:17   Log-Likelihood:                -62.746
No. Observations:                 100   AIC:                             127.5
Df Residuals:                      99   BIC:                             130.1
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [95.0% Conf. Int.]
------------------------------------------------------------------------------
y              0.4020      0.021     19.020      0.0